deluge/ui/web/json_api.py

```diff
@@ -251,10 +252,13 @@ def _on_json_request_failed(self, reason, request):
         return ""
 
     def _send_response(self, request, response):
-        response = json.dumps(response)
-        request.setHeader("content-type", "application/x-json")
-        request.write(compress(response, request))
-        request.finish()
+        try:
+            response = json.dumps(response)
+            request.setHeader("content-type", "application/x-json")
+            request.write(compress(response, request))
+            request.finish()
+        except TypeError, e:
```

Comment:
If the TypeError come from json.dumps then try should only wrap that with request in else. Also old style exception, needs `as ex` for develop branch.